### PR TITLE
 - remove issues pointed out by sonar in sbml dir

### PR DESCRIPTION
**Pull request by fbergmann**
## Description
removed sonar issues from 'sbml' dir

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated all documentation necessary.
- [ ] I have checked spelling in (new) comments.

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

File: src/sbml/Reaction.cpp [hunk 1]
```diff
@@ -1990,9 +2003,7 @@ Reaction::removeChildObject(const std::string& elementName, const std::string& i
 
   if (elementName == "kineticLaw")
   {
-    KineticLaw* t = getKineticLaw();
-    if (unsetKineticLaw() == LIBSBML_OPERATION_SUCCESS)
-      return t;
+    unsetKineticLaw(); // already deletes the kineticLaw, so nothing to do after
```
Comment (skeating) [inline src/sbml/Reaction.cpp:2006]: surely there should be a return here

File: src/sbml/SpeciesReference.cpp [hunk 2]
```diff
@@ -982,9 +982,7 @@ SpeciesReference::removeChildObject(const std::string& elementName, const std::s
 
   if (elementName == "stoichiometryMath")
   {
-    StoichiometryMath* t = getStoichiometryMath();
-    if (unsetStoichiometryMath() == LIBSBML_OPERATION_SUCCESS)
-      return t;
+    unsetStoichiometryMath();
```
Comment (skeating) [inline src/sbml/SpeciesReference.cpp:985]: return ?

**Review by skeating — CHANGES_REQUESTED** [review]

File: src/sbml/Reaction.cpp (hunk 1 above)
Comment (fbergmann) [reply src/sbml/Reaction.cpp:2006]: the return is at the end of the function ... so i didn't see a need to duplicate it here

File: src/sbml/SpeciesReference.cpp (hunk 2 above)
Comment (fbergmann) [reply src/sbml/SpeciesReference.cpp:985]: the return is at the end of the function ... so i didn't see a need to duplicate it here, can change it though

**Review by skeating — APPROVED** [review]

**Pull request merged**